src/models/StyledComponent.js

```diff
@@ -32,7 +32,8 @@ export default (ComponentStyle: Function, constructWithOptions: Function) => {
     identifiers[displayName] = nr
 
     const hash = ComponentStyle.generateName(displayName + nr)
-    return `${displayName}-${hash}`
+    const componentId = `${displayName}-${hash}`
+    return parentComponentId ? `${parentComponentId}-${componentId}` : componentId
```

Comment:
Maybe `parentComponentId !== undefined`? Sorry for the nit, but we have an ongoing perf pr and it'd be good to keep things optimised 😅